cypress/integration/save.cy.js

```diff
@@ -0,0 +1,188 @@
+import {
+    DIMENSION_ID_DATA,
+    isYearOverYear,
+    visTypeDisplayNames,
+} from '@dhis2/analytics'
+
+import { openDimension } from '../elements/DimensionsPanel'
+import {
+    selectDataElements,
+    clickDimensionModalUpdateButton,
+    selectRelativePeriods,
+} from '../elements/DimensionModal'
+import { changeVisType } from '../elements/VisualizationTypeSelector'
+import {
+    expectAOTitleToBeDirty,
+    expectAOTitleToBeUnsaved,
+    expectAOTitleToBeValue,
+    expectAOTitleToNotBeDirty,
+    expectVisualizationToBeVisible,
+} from '../elements/Chart'
+import { TEST_DATA_ELEMENTS } from '../utils/data'
+import {
+    //openRandomAOCreatedByOthers,
+    saveNewAO,
+    closeFileMenu,
+    saveAOAs,
+    saveExistingAO,
+    openAOByName,
+    deleteAO,
+    expectFileMenuButtonToBeEnabled,
+    expectFileMenuButtonToBeDisabled,
+    FILE_MENU_BUTTON_SAVEAS,
+    FILE_MENU_BUTTON_DELETE,
+    FILE_MENU_BUTTON_RENAME,
+    FILE_MENU_BUTTON_TRANSLATE,
+    FILE_MENU_BUTTON_SHARE,
+    FILE_MENU_BUTTON_GETLINK,
+} from '../elements/FileMenu'
+import { expectRouteToBeAOId, expectRouteToBeEmpty } from '../elements/Route'
+import { getRandomVisType } from '../utils/random'
+import {
+    clickMenuBarFileButton,
+    clickMenuBarUpdateButton,
+} from '../elements/MenuBar'
+import { selectYoyCategoryOption } from '../elements/Layout'
+import {
+    expectStartScreenToBeVisible,
+    goToStartPage,
+} from '../elements/StartScreen'
+
+const TEST_VIS_NAME = `TEST ${new Date().toLocaleString()}`
+const TEST_VIS_NAME_UPDATED = `${TEST_VIS_NAME} - updated`
+const TEST_VIS_DESCRIPTION = 'Generated by Cypress'
+const TEST_VIS_TYPE = getRandomVisType()
+const TEST_VIS_TYPE_NAME = visTypeDisplayNames[TEST_VIS_TYPE]
+
+// TODO: Add test to check that the description is saved and shown in the interpretations panel
+
+describe('saving an AO', () => {
+    describe('"save" and "save as" for a new AO', () => {
```

Comment:
These tests seem to break some Cypress best practices rules:
* https://docs.cypress.io/guides/references/best-practices.html#Having-tests-rely-on-the-state-of-previous-tests
* https://docs.cypress.io/guides/references/best-practices.html#Having-tests-rely-on-the-state-of-previous-tests

I tend to agree with the suggestions in these links. I think this would be better as a single test. As it stands now, they cannot be run in isolation.

Reply:
Indeed, very valid points. As mentioned in PMs I've now refactored most of these to better follow the best practice rules, however in some cases deviate to instead follow a `describe - describe - it` pattern (where the second `describe` acts like an `it` and the `it` is a subpart of a "test") to get a compromise between following the rules and having good usability / readability / ease of debugging for devs.